plugins/tech-insights-backend/src/service/fact/FactRetrieverEngine.test.ts

```diff
@@ -59,94 +55,169 @@ const testFactRetriever: FactRetriever = {
     ];
   }),
 };
-const cadence = '1 * * * *';
+const defaultCadence = '1 * * * *';
 describe('FactRetrieverEngine', () => {
   let engine: FactRetrieverEngine;
-  let factSchemaAssertionCallback: (
+  type FactSchemaAssertionCallback = (
     factSchemaDefinition: FactSchemaDefinition,
   ) => void;
-  let factInsertionAssertionCallback: (facts: TechInsightFact[]) => void;
 
-  const mockRepository: TechInsightsStore = {
-    insertFacts: (facts: TechInsightFact[]) => {
-      factInsertionAssertionCallback(facts);
-      return Promise.resolve();
-    },
-    insertFactSchema: (def: FactSchemaDefinition) => {
-      factSchemaAssertionCallback(def);
-      return Promise.resolve();
-    },
-  } as unknown as TechInsightsStore;
+  jest.setTimeout(15000);
 
-  const mockFactRetrieverRegistry: FactRetrieverRegistry = {
-    listRetrievers(): FactRetriever[] {
-      return [testFactRetriever];
-    },
-    listRegistrations(): FactRetrieverRegistration[] {
-      return [{ factRetriever: testFactRetriever, cadence }];
-    },
-  } as unknown as FactRetrieverRegistry;
+  type FactInsertionAssertionCallback = ({
+    facts,
+    id,
+  }: {
+    id: string;
+    facts: TechInsightFact[];
+  }) => void;
 
-  const defaultEngineConfig = {
-    factRetrieverContext: {
-      logger: getVoidLogger(),
-      config: ConfigReader.fromConfigs([]),
-      discovery: {
-        getBaseUrl: (_: string) => Promise.resolve('http://mock.url'),
-        getExternalBaseUrl: (_: string) => Promise.resolve('http://mock.url'),
+  function createMockRepository(
+    insertCallback: FactInsertionAssertionCallback,
+    assertionCallback: FactSchemaAssertionCallback,
+  ): TechInsightsStore {
+    return {
+      insertFacts: ({
```

Comment:
just a hint about async - this thing could be shortened to

```ts
async insertFacts(f: { facts: TechInsightFact[]; id: string }) {
  insertCallback(f);
}
```

ends up a bit clearer, need for using `Promise.resolve` is pretty rare. Similar below

Reply:
Gotcha, thanks! I'll clear these up a bit then